backend/remote-state/kubernetes/backend_test.go

```diff
@@ -67,7 +73,57 @@ func TestBackendLocks(t *testing.T) {
 	backend.TestBackendStateForceUnlock(t, b1, b2)
 }
 
-func cleanupK8sSecrets(t *testing.T) {
+func TestBackendLocksSoak(t *testing.T) {
+	testACC(t)
+	defer cleanupK8sResources(t)
+
+	clientCount := 1000
+	lockCount := 0
+
+	lockers := []statemgr.Locker{}
+	for i := 0; i < clientCount; i++ {
+		b := backend.TestBackendConfig(t, New(), backend.TestWrapConfig(map[string]interface{}{
+			"secret_suffix": secretSuffix,
+		}))
+
+		s, err := b.StateMgr(backend.DefaultStateName)
+		if err != nil {
+			t.Fatalf("Error creating state manager: %v", err)
+		}
+
+		lockers = append(lockers, s.(statemgr.Locker))
+	}
+
+	wg := sync.WaitGroup{}
+	for i, l := range lockers {
+		wg.Add(1)
+		go func(locker statemgr.Locker, i int) {
+			r := rand.Intn(10)
+			time.Sleep(time.Duration(r) * time.Microsecond)
+			li := state.NewLockInfo()
+			li.Operation = "test"
+			li.Who = fmt.Sprintf("client-%v", i)
+			_, err := locker.Lock(li)
+			if err == nil {
+				t.Logf("[INFO] Client %v got the lock\r\n", i)
+				lockCount++
+			}
+			wg.Done()
+		}(l, i)
+	}
+
+	wg.Wait()
+
+	if lockCount > 1 {
```

Comment:
We can't easily count multiple successful locks and unlocks, since even atomic operations would not be perfectly coordinated with the lock and unlock operations, so we need to rely on checking the lock integrity itself. The best method we have for that is ensuring `Unlock` never returns an error. If multiple clients were locked, then one of those must fail when the wrong ID is used to unlock the state.